QAChat/Data_Processing/confluence_preprocessor.py

```diff
@@ -0,0 +1,248 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2023 Hafidz Arifin
+# SPDX-FileCopyrightText: 2023 Abdelkader Alkadour
+
+
+from atlassian import Confluence
+from dotenv import load_dotenv
+from bs4 import BeautifulSoup
+from datetime import datetime
+import os, io, requests, PyPDF2, supabase, hashlib
+
+
+# Load environment variables
+load_dotenv("../tokens.env")
+
+# Get Confluence API credentials from environment variables
+CONFLUENCE_ADRESS = os.getenv("CONFLUENCE_ADRESS")
+CONFLUENCE_USERNAME = os.getenv("CONFLUENCE_USERNAME")
+CONFLUENCE_PASSWORD = os.getenv("CONFLUENCE_PASSWORD")
+
+# Get Supabase API credentials from environment variables
+SUPABASE_URL = os.getenv("SUPABASE_URL")
+SUPABASE_SERVICE_KEY = os.getenv("SUPABASE_SERVICE_KEY")
+
+
+class ConfluencePreprocessor:
+    spaces = []
+    pages = []
```

Comment:
This class should inherit from DataPreprocessor. so the constructor should look something like:
`classConfluencePreprocessor(DataPreprocessor):`
and then it should implement the method:
`def load_preprocessed_data(self, before: datetime, after: datetime) -> List[DataInformation]:`
When you have this, you can use the document embedded to create the embeddings and save it into the supabase database. Therefore you just need to add ypir preprocessor to the function:
`def store_information_in_database(self, typ: DataSource):` line 43 in the same way the DummyPrepocessor was added and then you can call it something like in the main.py of the Data_Processing folder.